### ISSUE TITLE
Unable to import workflow

### ISSUE THREAD

**Issue by dleehr** [Duke-GCB/bespin-workflows#167]
I'm attempting to import the [v0.9.4/exomeseq-gatk4-preprocessing.cwl](https://github.com/Duke-GCB/bespin-cwl/releases/download/v0.9.4/exomeseq-gatk4-preprocessing.cwl) workflow from [exomeseq-gatk4-preprocessing-bespin-dev.json](https://github.com/Duke-GCB/bespin-cwl/blob/727ad5dbd15ef6aaa9cd5a3f7b23e74ff74663ea/examples/exome-seq/exomeseq-gatk4-preprocessing-bespin-dev.json) using the scripts in bespin-cwl

```
$ ./scripts/post_questionnaire.sh examples/exome-seq/exomeseq-gatk4-preprocessing-bespin-dev.json
{"detail":"Unable to import workflow"}
```

In the bespin-api logs, I see the following:

```
    "No cwlVersion found. "
ValidationException: No cwlVersion found. Use the following syntax in your CWL document to declare the version: cwlVersion: <version>.
Note: if this is a CWL draft-2 (pre v1.0) document then it will need to be upgraded first.
```

This is happening both on the dev server and on my local development instance.

**Comment by dleehr** [comment]
Reproducing this issue with other workflows isn't 100% reliable, and seems to depend on if the CWL URL has been fetched by the process before. That would suggest that the caching used by schema-salad isn't working as expected, but it's certainly not the same issue as [last time](https://github.com/common-workflow-language/schema_salad/issues/158)

**Comment by dleehr** [comment]
I built up a local docker environment using docker-compose, and find that this workflow import succeeds the first time but fails the second. I do suspect it's caching beneath schema salad.

**Comment by dleehr** [comment]
Reduced test case:

```python
from cwltool.context import LoadingContext
from cwltool.workflow import default_make_tool
from cwltool.resolver import tool_resolver
from cwltool.load_tool import load_tool

url = 'https://github.com/Duke-GCB/bespin-cwl/releases/download/v0.9.4/exomeseq-gatk4-preprocessing.cwl'
context1 = LoadingContext({"construct_tool_object": default_make_tool,
                           "resolver": tool_resolver,
                           "disable_js_validation": True})
print('Calling load_tool first time')
parsed1 = load_tool(url + '#main', context1)
print('Calling load_tool second time')
# this fails
parsed2 = load_tool(url + '#main', context1) 
```

**Comment by dleehr** [comment]
May be related to #155

**Comment by dleehr** [comment]
Exception is raised on https://github.com/common-workflow-language/cwltool/blob/947bb587252de9cce9b309898f0b5a34a24b56ac/cwltool/load_tool.py#L240

Updated code:

```python
from cwltool.context import LoadingContext
from cwltool.workflow import default_make_tool
from cwltool.resolver import tool_resolver
from cwltool.load_tool import load_tool

def parse(url):
    context = LoadingContext({"construct_tool_object": default_make_tool,
                               "resolver": tool_resolver})
    parsed = load_tool(url + '#main', context)
    return parsed

# a packed workflow
packed_url = 'https://github.com/Duke-GCB/bespin-cwl/releases/download/v0.9.4/exomeseq-gatk4-preprocessing.cwl'

parse(packed_url)
parse(packed_url)
```

**Comment by dleehr** [comment]
So here's the current status as it relates to bespin-api and loading workflows

- I run `post_questionnaire.sh` with a JSON file referencing a CWL url and other JobQuestionnaire details.
- the bespin-api python process receives this POST and uses `cwltool.load_tool` to fetch and load the referenced packed CWL workflow from the GitHub release
- The first time this URL is encountered by the process, the parsing succeeds. So if all the JSON fields are filled out and there is no workflow version conflict, the questionnaire is successfully created
- However, if I have to tweak the JSON and re-POST it, the import immediately fails. The bespin-api python process will no longer fetch/load that packed workflow successfully, and it will log the `No cwlVersion found` error
- If I restart the docker container running bespin-api, the next attempt to import the workflow will successfully fetch/load the packed workflow

So long story short - In the event I need to tweak JSON and make a second attempt at loading a packed workflow, I need to restart bespin-api.

**Comment by dleehr** [comment]
Closing this issue, as we no longer load workflows this way.
